### PR TITLE
tableauserverclient 0.30 :snowflake:

### PR DESCRIPTION
**Pull request by lorepirri**
tableauserverclient 0.30 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-3725]
- dev_url: https://github.com/tableau/server-client-python/tree/v0.30
- conda-forge: https://github.com/conda-forge/tableauserverclient-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/tableauserverclient
- Relevant dependency PRs:
  - AnacondaRecipes/urllib3-feedstock#18
  - AnacondaRecipes/requests-feedstock#12
  - tests AnacondaRecipes/requests-mock-feedstock#2

### Explanation of changes:

- update version, fix recipe

[PKG-3725]: https://anaconda.atlassian.net/browse/PKG-3725?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lorepirri** [comment]
It is required `requests>=2.31.0`. The latest `requests==2.31.0`, which is also the latest version, in our index requires `urllib3>=1.21.1,<2` (instead of the `urllib3 >=1.21.1,<3` [specified in the upstream](https://github.com/psf/requests/blob/147c8511ddbfa5e8f71bbf5c18ede0c4ceb3bba4/setup.py#L61-L66), because [we made a conservative choice here](https://github.com/AnacondaRecipes/requests-feedstock/pull/11#discussion_r1274522216) @skupr-anaconda ).

At the same time, in this recipe, the exact `urllib3==2.0.7` is needed, which clashes with the previous requirement.

@JeanChristopheMorinPerso what is your opinion? Shall we relax our `requests==2.31.0` from `urllib3>=1.21.1,<2` to `urllib3 >=1.21.1,<3`?

**Comment by JeanChristopheMorinPerso** [comment]
@lorepirri `requests` should be fixed. https://inspector.pypi.io/project/requests/2.31.0/packages/70/8e/0e2d847013cb52cd35b38c009bb167a1a26b2ce6cd6965bf26b47bc0bf44/requests-2.31.0-py3-none-any.whl/requests-2.31.0.dist-info/METADATA#line.35.

**Comment by lorepirri** [comment]
> @lorepirri `requests` should be fixed. https://inspector.pypi.io/project/requests/2.31.0/packages/70/8e/0e2d847013cb52cd35b38c009bb167a1a26b2ce6cd6965bf26b47bc0bf44/requests-2.31.0-py3-none-any.whl/requests-2.31.0.dist-info/METADATA#line.35.

Thanks! In progress at AnacondaRecipes/requests-feedstock#12

File: recipe/meta.yaml [hunk 1]
```diff
@@ -36,16 +35,16 @@ test:
...
     - requests-mock >=1.0,<2.0
+  source_files:
+    - test
+  commands:
+    - pip check
+    # ignore completely tests that depend on data files not contained in the tarball
+    - pytest test --ignore test/test_workbook.py --ignore test/test_view.py --ignore test/test_flow.py --ignore test/test_fileuploads.py --ignore test/test_filesys_helpers.py --ignore test/test_datasource.py
 
 about:
   home: https://github.com/tableau/server-client-python
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:51]: The source code also has a vendored `Versioneer` under the `CC0-1.0` license, see https://github.com/tableau/server-client-python/blob/v0.30/LICENSE.versioneer. And this license file should be included too

Comment (lorepirri) [reply recipe/meta.yaml:51]: addressed

**Review by skupr-anaconda — APPROVED** [review]

**Comment by JeanChristopheMorinPerso** [comment]
@skupr-anaconda versioneer will not be in the final artifact. Versioneer generates a `_version.py` file, see https://inspector.pypi.io/project/tableauserverclient/0.30/packages/0b/a6/23eed1b99ddd7d1cddde0f77fbf01aed8f0892a1e8f3dd47807358cb7965/tableauserverclient-0.30-py3-none-any.whl/tableauserverclient/_version.py.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -36,22 +35,24 @@ test:
...
-  license: MIT
-  license_family: MIT
-  license_file: LICENSE
+  license: MIT and CC0-1.0
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: That's wrong. The license is MIT. Versioneer is not included in the file artifacts.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -36,22 +35,24 @@ test:
...
+  license_family: OTHER
+  license_file: 
+    - LICENSE
+    - LICENSE.versioneer
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (lorepirri) [reply recipe/meta.yaml]: ok

File: recipe/meta.yaml (hunk 3 above)
Comment (lorepirri) [reply recipe/meta.yaml]: ok

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,33 +1,32 @@
...
-  build:
-    - pytest-runner
-    - versioneer >=0.24
   host:
-    - python >=3.7
+    - python
     - pip
-    - pytest-runner
+    - wheel
+    - versioneer >=0.29
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:23]:
```suggestion
    - versioneer >=0.29
    - setuptools >=68.0
```

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Pull request merged**